### ISSUE TITLE
minSDK issue while using the package

### ISSUE THREAD

**Issue by pratik037** [Sh1d0w/multi_image_picker#170]
**Bug Report**

`multi_image_picker` version:   ^4.3.3

Is it happening on Android:  Yes

Is it happening on iOS:  Can't Test

```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel beta, v1.6.3, on Linux, locale en_GB.UTF-8)
 
[✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
[✓] Android Studio (version 3.4)
[✓] IntelliJ IDEA Community Edition (version 2019.1)
[✓] VS Code (version 1.34.0)
[✓] Connected device (1 available)

• No issues found!
```

**Describe the bug**
I think this will explain it better?
```
AndroidManifest.xml Error:
	uses-sdk:minSdkVersion 16 cannot be smaller than version 19 declared in library [:multi_image_picker] /home/pratik037/Flutter Projects/memories/build/multi_image_picker/intermediates/library_manifest/debug/AndroidManifest.xml as the library might be using APIs not available in 16
	Suggestion: use a compatible library with a minSdk of at most 16,
		or increase this project's minSdk version to at least 19,
		or use tools:overrideLibrary="com.vitanov.multiimagepicker" to force usage (may lead to runtime failures)
```

**Comment by gzf6** [comment]
same issue, and when I change the minSdkVersion from 16 to 19, the bug info disappear but the app will crash when I open it.

**Comment by pratik037** [comment]
> same issue, and when I change the minSdkVersion from 16 to 19, the bug info disappear but the app will crash when I open it.

Waiting for the developer to reply

**Comment by Sh1d0w** [comment]
As the message says, the min SDK version needs to be set to 19 in order to use the plugin, you can change it in the build.gradle

What error do you get if you change it to 19?

**Comment by pratik037** [comment]
> As the message says, the min SDK version needs to be set to 19 in order to use the plugin, you can change it in the build.gradle
> 
> What error do you get if you change it to 19?

The App crashes when the changes are made

**Comment by Sh1d0w** [comment]
@pratik037 I understand that but how I can help you without knowing why it crashes? 

Please open the terminal and run `adb logcat` and then run the app again and capture the error message.

I will close this as it is not reproducible. If you provide the error you get I may be able to help you then.

**Comment by Sh1d0w** [comment]
PS @pratik037 check if this solves your problem 

https://github.com/Sh1d0w/multi_image_picker/issues/165#issuecomment-498928371

**Comment by pratik037** [comment]
> PS @pratik037 check if this solves your problem
> 
> [#165 (comment)](https://github.com/Sh1d0w/multi_image_picker/issues/165#issuecomment-498928371)

Although I had already refactored the code, but let me try again and update here in a while

**Comment by summer789** [comment]
I also encountered this problem. Is there a solution now?

**Comment by Sh1d0w** [comment]
@summer789 When you are saying you have a problem, provide more information. How I can help you when I don't know what is the problem?

Make sure you have migrated to android x and you have min SDK set to 19 in your android manifest xml.

For example app you can see the code in the example folder in this repo and run it.

**Comment by pratik037** [comment]
@Sh1d0w hey man, I tried the package and changed the minSDK version.
But now if I want to choose image, its throwing this error
```
I/Timeline(25152): Timeline: Activity_launch_request time:15183012 intent:Intent { act=android.content.pm.action.REQUEST_PERMISSIONS pkg=com.google.android.packageinstaller (has extras) }
```

Can you help in understanding the error please?

**Comment by summer789** [comment]
@Sh1d0w   I tried it again today. After upgrading to andriodX, using vscode, it is normal, thank you.

**Comment by Sh1d0w** [comment]
@pratik037 Did you do the initial setup as described in the docs https://sh1d0w.github.io/multi_image_picker/#/initialsetup ?

**Comment by pratik037** [comment]
> @pratik037 Did you do the initial setup as described in the docs https://sh1d0w.github.io/multi_image_picker/#/initialsetup ?

Yes, I did. Also, the error is solved. Can you tell me why grid builder is having error?
I mean when I call buildGridView, renderbox error occurs

**Comment by pratik037** [comment]
Here's the code: 
``` Dart
import 'package:flutter/material.dart';
import 'package:flutter/services.dart';
import 'package:memories/pick_image.dart';
import 'package:multi_image_picker/multi_image_picker.dart';
import 'dart:async';

class AddDetails extends StatefulWidget {
  final String place_id;
  final String name;
  AddDetails(this.place_id, this.name);

  @override
  _AddDetailsState createState() => _AddDetailsState();
}

class _AddDetailsState extends State<AddDetails> {
  FocusNode myFocusNode;
  TextEditingController _editingController = TextEditingController();
   List<Asset> images = List<Asset>();


  @override
  void initState() {
    super.initState();
    myFocusNode = FocusNode();
  }

  @override
  void dispose() {
    myFocusNode.dispose();
    super.dispose();
  }

  @override
  Widget build(BuildContext context) {
    final bool showFab = MediaQuery.of(context).viewInsets.bottom == 0.0;
    final bool showbtn = MediaQuery.of(context).viewInsets.bottom == 0.0;
    return Scaffold(
        body: GestureDetector(
          onTap: () {
            FocusScope.of(context).requestFocus(myFocusNode);
          },
          child: SingleChildScrollView(
            child: Column(
              children: <Widget>[
                SizedBox(height: 20),
                Text(
                  widget.name,
                  style: TextStyle(
                      fontSize: 30,
                      fontWeight: FontWeight.bold,
                      fontFamily: 'Pacifico'),
                  textAlign: TextAlign.center,
                ),
                SizedBox(height: 5),
                Text(
                  "Lets add some memories, excited?",
                  style: TextStyle(fontFamily: 'Montserrat'),
                ),
                Padding(
                  padding: const EdgeInsets.all(8.0),
                  child: TextFormField(
                    focusNode: myFocusNode,
                    decoration: InputDecoration(
                      border: OutlineInputBorder(),
                      hintText: 'Only best of the memories....',
                      hintStyle:
                          TextStyle(fontFamily: 'Dancing Script', fontSize: 22),
                      labelText: 'Description',
                      labelStyle:
                          TextStyle(fontFamily: 'Dancing Script', fontSize: 25),
                    ),
                    // autofocus: true,
                    autocorrect: true,
                    controller: _editingController,
                    minLines: 1,
                    maxLines: 20,
                    scrollPadding: EdgeInsets.all(8),
                  ),
                ),
                Padding(
                  padding: const EdgeInsets.all(8.0),
                  child: images.length == null
                      ? Text('Add Images to your memory')
                      : Column(
                          children: <Widget>[
                            Text(
                              " Images Selected ",
                              textAlign: TextAlign.left,
                            ),
                            SizedBox(
                              height: 5,
                            ),
                            Expanded(
                              child: buildGridView(),
                            )
                            
                          ],
                        ),
                ),
                Center(
                  child: Padding(
                    padding: const EdgeInsets.all(8.0),
                    child: showbtn
                        ? FloatingActionButton.extended(
                            heroTag: 'unique2',
                            focusNode: myFocusNode,
                            onPressed: () {
                              final eSnackBar = SnackBar(content: Text("Add some text"),);
                              final aSnackBar = SnackBar(content: Text("Added to memories successfully"),);
                              _editingController.text.trim().isEmpty? Scaffold.of(context).showSnackBar(eSnackBar) :
                              Scaffold.of(context).showSnackBar(aSnackBar);
                            },
                            backgroundColor: _editingController.text.isNotEmpty
                                ? Colors.green
                                : Colors.grey,
                            icon: Icon(Icons.done),
                            label: Text('Add to memories'),
                            elevation: 2,
                          )
                        : null,
                  ),
                ),
              ],
            ),
          ),
        ),
        floatingActionButton: Padding(
          padding: const EdgeInsets.all(4.0),
          child: showFab
              ? FloatingActionButton.extended(
                  heroTag: 'unique1',
                  label: Text('Add Images'),
                  icon: Icon(Icons.add),
                  onPressed: (){
                    loadAssets();
                  },
                  elevation: 3,
                  backgroundColor: Colors.black,
                  focusNode: myFocusNode,
                )
              : null,
        ),
      );
  }

  Widget buildGridView() {
    return GridView.count(
      shrinkWrap: true,
      crossAxisCount: 2,
      children: List.generate(images.length, (index) {
        Asset asset = images[index];
        return AssetThumb(
          asset: asset,
          width: 300,
          height: 300,
        );
      }),
    );
  }
Future<void> deleteAssets() async {
    await MultiImagePicker.deleteImages(assets: images);
    setState(() {
      images = List<Asset>();
    });
  }

  Future<void> loadAssets() async {
    List<Asset> resultList = List<Asset>();
    String error = 'No Error Dectected';

    try {
      resultList = await MultiImagePicker.pickImages(
          maxImages: 300,
          enableCamera: true,
          selectedAssets: images,
          );
    } on PlatformException catch (e) {
      error = e.message;
    }

    // If the widget was removed from the tree while the asynchronous platform
    // message was in flight, we want to discard the reply rather than calling
    // setState to update our non-existent appearance.
    if (!mounted) return;

    setState(() {
      images = resultList;
    });
  }

  // Widget showImages(){
  //   return GridView.builder(
  //     itemCount: images.length,
  //     shrinkWrap: true,
  //     gridDelegate: SliverGridDelegateWithFixedCrossAxisCount(crossAxisCount: 2),
  //     itemBuilder: (BuildContext context, int index) {
  //       Asset asset = images[index];
  //       return Padding(
  //         padding: EdgeInsets.all(4),
  //         child: AssetThumb(
  //         asset: asset,
  //         width: 300,
  //         height: 300,
  //       ),
  //       );
  //     },
  //   );
  // }

}
```

**Comment by pratik037** [comment]
Added in gist instead. 
[Line 95](https://gist.github.com/pratik037/667c58f557467c4aedc00fbe7b7f65ad) has the implementation.

**Comment by jpainam** [comment]
Hi, Flutter support API down to level 16. I think it will be good for this plugin to follow this trends. I have more than 20 plugins. this is the first plugin that requires changing the minsdk while i'm trying to target API Level 16.  Kindly provide support for API Level 16
```dart
Manifest merger failed : uses-sdk:minSdkVersion 16 cannot be smaller than version 19 declared in library [:multi_image_picker] D:\flutter\workspace\maraka\build\multi_image_picker\intermediates\library_manifest\debug\AndroidManifest.xml as the library might be using APIs not available in 16
  	Suggestion: use a compatible library with a minSdk of at most 16,
  		or increase this project's minSdk version to at least 19,
  		or use tools:overrideLibrary="com.vitanov.multiimagepicker" to force usage (may lead to runtime failures)

```

**Comment by Sh1d0w** [comment]
@jpainam There was a time when this plugin used another picker library for android called Matisse, and it had minSdkVersion 19. It was left like this when I changed the libraries.

If you can open a PR to change that, and also make sure it works fine and test it, I will accept the PR.

**Comment by jpainam** [comment]
I have submitted PR #399 . Tested on Android API Level 16. It's working fine.  maybe more testing is needed.
